File: frontend/src/models/WarehouseRelease.ts
import type { WarehouseReleaseItem } from "./WarehouseReleaseItem";

export interface WarehouseRelease {
  id: number;
  releaseReference: string;
  releaseDate: string;
  destination: string;
  grandTotal: number;

  items: WarehouseReleaseItem[];
}

File: Models/WarehouseRelease.cs
using System.ComponentModel.DataAnnotations;

namespace JelycoWarehouse.Models
{
    public class WarehouseRelease
    {
        public int Id { get; set; }

        public string ReleaseReference { get; set; } = string.Empty;

        public DateTime ReleaseDate { get; set; }

        // Where the items are going
        [MaxLength(150)]
        public string Destination { get; set; } = string.Empty;

        public decimal GrandTotal { get; set; }

        public ICollection<WarehouseReleaseItem> Items { get; set; }
            = new List<WarehouseReleaseItem>();

        public ICollection<Transaction> Transactions { get; set; }
            = new List<Transaction>();
    }
}